storage/innobase/srv/srv0srv.cc

```diff
@@ -3600,3 +3610,47 @@ srv_is_undo_tablespace(
 	       && space_id < (srv_undo_space_id_start
 			      + srv_undo_tablespaces_open));
 }
+
+/** Gets page fragmentation statistics
+@param[out]	stats	a reference to fragmentation statistics to fill */
+UNIV_INTERN
+void
+srv_get_fragmentation_stats(fragmentation_stats_t& stats)
+{
+	THD*	thd = current_thd;
+	if (UNIV_LIKELY(thd)) {
+		stats.scan_pages_contiguous =
+			thd->status_var.scan_pages_contiguous;
+		stats.scan_pages_disjointed =
+			thd->status_var.scan_pages_disjointed;
+		stats.scan_pages_total_seek_distance =
+			thd->status_var.scan_pages_total_seek_distance;
+		stats.scan_data_in_pages =
+			thd->status_var.scan_data_in_pages;
+		stats.scan_garbage_in_pages =
+			thd->status_var.scan_garbage_in_pages;
+	} else {
+		stats = fragmentation_stats_t();
```

Comment:
A very minor comment, optional to address: for a POD I'd just memset instead of construct and assign.

Reply:
fixed